src/Engine/ProtoCore/Lang/FunctionGroup.cs

```diff
@@ -148,10 +150,19 @@ public void CopyPublic(List<FunctionEndPoint> rhs)
                     continue;
                 }
 
-                if (fep.DoesTypeDeepMatch(reducedParamSVs, runtimeCore))
+                bool typesOK = true;
+                foreach (List<StackValue> reducedParamSVs in allReducedParamSVs)
                 {
-                    ret.Add(fep);
+                    if (!fep.DoesTypeDeepMatch(reducedParamSVs, runtimeCore))
```

Comment:
YEa seems to be righter this time. Previously it seemed to only check a match for the first param